nrf_wifi/hw_if/hal/src/hal_api.c

```diff
@@ -1256,22 +1289,11 @@ void nrf_wifi_hal_dev_rem(struct nrf_wifi_hal_dev_ctx *hal_dev_ctx)
 {
 	unsigned int i = 0;
 
-
-	nrf_wifi_bal_dev_rem(hal_dev_ctx->bal_dev_ctx);
-
-	nrf_wifi_osal_mem_free(hal_dev_ctx->hpriv->opriv,
-			       hal_dev_ctx->tx_buf_info);
-	hal_dev_ctx->tx_buf_info = NULL;
-
-	for (i = 0; i < MAX_NUM_OF_RX_QUEUES; i++) {
-		nrf_wifi_osal_mem_free(hal_dev_ctx->hpriv->opriv,
-				       hal_dev_ctx->rx_buf_info[i]);
-		hal_dev_ctx->rx_buf_info[i] = NULL;
-	}
-
 	nrf_wifi_osal_tasklet_kill(hal_dev_ctx->hpriv->opriv,
 				   hal_dev_ctx->event_tasklet);
 
+	hal_rpu_eventq_drain(hal_dev_ctx);
```

Comment:
Use return value to log an error?

Reply:
There is error returned currently, may be I will just make it void.